generators/server/templates/src/main/java/package/config/_WebConfigurer.java

```diff
@@ -53,13 +54,13 @@ public void onStartup(ServletContext servletContext) throws ServletException {
         EnumSet<DispatcherType> disps = EnumSet.of(DispatcherType.REQUEST, DispatcherType.FORWARD, DispatcherType.ASYNC);<% if (clusteredHttpSession == 'hazelcast') { %>
         initClusteredHttpSessionFilter(servletContext, disps);<% } %>
         initMetrics(servletContext, disps);
-        if (env.acceptsProfiles(Constants.SPRING_PROFILE_PRODUCTION)) {
-            initCachingHttpHeadersFilter(servletContext, disps);
-            initStaticResourcesProductionFilter(servletContext, disps);
-        }<% if (devDatabaseType == 'h2Disk' || devDatabaseType == 'h2Memory') { %>
-        if (env.acceptsProfiles(Constants.SPRING_PROFILE_DEVELOPMENT)) {
-            initH2Console(servletContext);
-        }<% } %>
+        if (env.acceptsProfiles(Constants.SPRING_PROFILE_PRODUCTION)) {<% if (!skipClient) { %>
```

Comment:
Yes you are right that I would disagree. For me the overall benefit is just
not having some extra baggage in the bundled war at the end (Im not sure
that the sourcemap mangling issue will go away with this as it seems like a
browser restriction).

But the cost is to run a gulp task always and to loose the awesome chrome
dev tool file system mapping feature. So this also would mean we make gulp
an even tighter dependency which im not very comfy with.

So for me the cost seems more than benefit.

But as you said let others comment as well.

About node_modules its not really required to start development if you skip
using gulp serve, where as bower files are required. I mean in my last
project my entire dev team was behind extreme corporate proxy so they
couldn't use bower or node, i was the only one who had access to run those
and my team would checkout bower files from git and work and of course they
didnt use gulp as well. I was the only one doing prod builds anyway.

There could be so many such cases where a checked in ready to run codebade
would be required.

And that's the reason we even provide a default app.constants.js as well
though they will be overwritten by gulp when run. The idea is to make it as
easy as possible for any team to start development even if they are behind
such firewalls and stuff.
On 21 Feb 2016 20:17, "Erik Kemperman" <notifications@github.com> wrote:

> In
> generators/server/templates/src/main/java/package/config/_WebConfigurer.java
> <https://github.com/jhipster/generator-jhipster/pull/2945#discussion_r53565757>
> :
>
> > @@ -53,13 +54,13 @@ public void onStartup(ServletContext servletContext) throws ServletException {
> >          EnumSet<DispatcherType> disps = EnumSet.of(DispatcherType.REQUEST, DispatcherType.FORWARD, DispatcherType.ASYNC);<% if (clusteredHttpSession == 'hazelcast') { %>
> >          initClusteredHttpSessionFilter(servletContext, disps);<% } %>
> >          initMetrics(servletContext, disps);
> > -        if (env.acceptsProfiles(Constants.SPRING_PROFILE_PRODUCTION)) {
> > -            initCachingHttpHeadersFilter(servletContext, disps);
> > -            initStaticResourcesProductionFilter(servletContext, disps);
> > -        }<% if (devDatabaseType == 'h2Disk' || devDatabaseType == 'h2Memory') { %>
> > -        if (env.acceptsProfiles(Constants.SPRING_PROFILE_DEVELOPMENT)) {
> > -            initH2Console(servletContext);
> > -        }<% } %>
> > +        if (env.acceptsProfiles(Constants.SPRING_PROFILE_PRODUCTION)) {<% if (!skipClient) { %>
>
> No worries, I didn't! I'm quite happy with the feedback, actually --
> obviously I really desperately need the insight of jhipster veterans for
> something like this!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jhipster/generator-jhipster/pull/2945/files#r53565757>
> .
>